homeassistant/components/ipma/weather.py

```diff
@@ -89,10 +90,33 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
     """Add a weather entity from a config_entry."""
     latitude = config_entry.data[CONF_LATITUDE]
     longitude = config_entry.data[CONF_LONGITUDE]
+    mode = config_entry.data[CONF_MODE]
 
     api = await async_get_api(hass)
     location = await async_get_location(hass, api, latitude, longitude)
 
+    # Migrate old unique_id
+    @callback
+    def _async_migrator(entity_entry: entity_registry.RegistryEntry):
+        # Reject if new unique_id
+        if any([mode in entity_entry.unique_id for mode in FORECAST_MODE]):
```

Comment:
This is a poor check. Since you know the format of the unique ID, you can extract it. 

You can also just count the number of commas' ? 

```suggestion
        if entity_entry.unique_id.count(',') == 2:
```